sound/soc/sof/debug-dsp-ops.c

```diff
@@ -0,0 +1,119 @@
+// SPDX-License-Identifier: GPL-2.0-only
+//
+// Copyright(c) 2024 Intel Corporation.
+//
+
+#include <linux/debugfs.h>
+#include <sound/sof/debug.h>
+#include "sof-priv.h"
+#include "ops.h"
+
+static ssize_t sof_dsp_ops_tester_dfs_read(struct file *file, char __user *buffer,
+					   size_t count, loff_t *ppos)
+{
+	struct snd_sof_dfsentry *dfse = file->private_data;
+	struct snd_sof_dev *sdev = dfse->sdev;
+	struct dentry *dentry;
+	const char *string;
+	size_t size_ret;
+
+	/* return the FW filename or path */
+	dentry = file->f_path.dentry;
+	if (!strcmp(dentry->d_name.name, "fw_filename"))
+		string = sdev->test_profile.fw_name;
+	else if (!strcmp(dentry->d_name.name, "fw_path"))
+		string = sdev->test_profile.fw_path;
```

Comment:
You are planning to extend this to libraries and topologies? Libraries likely.
Just thinking if we could get away with a single 'firmware' string which includes the path and the name.

Reply:
Yes libraries and topologies both. But as far as the topology is concerned, the test_profile can also be used to set the topology filename and path to test. I'll have to think about the library test when I get to it later